[Extensions.AWS] Enable X-Ray ALB trace context parsing with only Root value (#4403)
open-telemetry/opentelemetry-dotnet-contrib: src/OpenTelemetry.Extensions.AWS/CHANGELOG.md

* Fix sampling behaviour to be compatible with .NET 11.
  ([#4396](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/4396))

> * Changed AWSXRayPropagator behavior to parse AWS ALB headers with only Root value in it.

Reviewer: ```suggestion
* `AWSXRayPropagator` now parses AWS ALB headers with only a Root value.
```
